lib/src/react16_suggestors/pubspec_react_upgrader.dart

```diff
@@ -24,36 +26,71 @@ import 'package:source_span/source_span.dart';
 /// the file, it will be added.
 class PubspecReactUpdater implements Suggestor {
   /// Regex that matches the dependency constraint declaration for react.
-  static final RegExp reactDep = RegExp(
-    r'''^\s*react:\s*["']?([\d\s<>=^.]+)["']?\s*$''',
-    multiLine: true,
-  );
+  static final RegExp reactDep = reactDependencyRegExp;
 
-  /// Regex that matches the `dependencies:` key in a pubspec.yaml.
-  static final RegExp dependenciesKey = RegExp(
-    r'^\s*dependencies:$',
-    multiLine: true,
-  );
+  /// Regex that matches the dependency pubspec.yaml key.
+  static final RegExp dependenciesKey = dependencyRegExp;
 
+  /// Constraint to update react to.
   final VersionRange targetConstraint;
 
-  PubspecReactUpdater(this.targetConstraint);
+  /// Whether or not the dependency should be added if it is not already
+  /// present.
+  final bool shouldAddDependencies;
+
+  PubspecReactUpdater(this.targetConstraint,
+      {this.shouldAddDependencies = true});
 
   @override
   Iterable<Patch> generatePatches(SourceFile sourceFile) sync* {
     final contents = sourceFile.getText(0);
     final reactMatch = reactDep.firstMatch(contents);
+    bool mightNeedYamlEscaping(String scalarValue) =>
```

Comment:
#nit Can we break this out of this function? Either as a private method, private static method, or private top level function?